avocado/core/test.py

```diff
@@ -276,12 +276,15 @@ def record_and_warn(*args, **kwargs):
             self.__base_logdir_tmp = tempfile.TemporaryDirectory(prefix=prefix)
             self.__base_logdir = self.__base_logdir_tmp.name
 
-        self.__base_logdir = os.path.join(self.__base_logdir, 'test-results')
-        logdir = os.path.join(self.__base_logdir, self.name.str_filesystem)
-        if os.path.exists(logdir):
-            raise exceptions.TestSetupFail("Log dir already exists, this "
-                                           "should never happen: %s"
-                                           % logdir)
+        if self._config.get("run.test_runner", 'nrunner') == 'nrunner':
+            logdir = self.__base_logdir
+        else:
+            self.__base_logdir = os.path.join(self.__base_logdir, 'test-results')
+            logdir = os.path.join(self.__base_logdir, self.name.str_filesystem)
+            if os.path.exists(logdir):
+                raise exceptions.TestSetupFail("Log dir already exists, this "
+                                               "should never happen: %s"
+                                               % logdir)
```

Comment:
Also, since legacy runner is deprecated and not the default one, I think maybe we don't need this logic anymore.

Reply:
Because the `test.py` has a lot of legacy code, I decided to leave it here for now. And we can remove it when we will be removing every legacy code from `test.py`